libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/statistics/KotlinBuildStatsService.kt

```diff
@@ -106,21 +105,26 @@ internal abstract class KotlinBuildStatsService internal constructor() : BuildAd
                             )
                             instance = JMXKotlinBuildStatsService(mbs, beanName)
                         } else {
+                            val kotlinBuildStatProvider = project.provider{ KotlinBuildStatListener(beanName) }
                             val newInstance = DefaultKotlinBuildStatsService(gradle, beanName)
+
+                            if (isConfigurationCacheAvailable(gradle)) {
```

Comment:
:x: the check should be `gradleVersion >= 6.1` instead where `BuildEventsListenerRegistry` is available

Reply:
New listener is not ready for configuration cache